### PR TITLE
Fix cpu/memory metrix for cgroup2, fixes kadalu/kadalu#810

### LINKED ISSUE #810: [Bug]: kadalu_memory_usage_in_bytes is -1 in Openshift/OKD
**Describe the bug**
On OKD4.10 (but should be valid for Openshift/OKD 4.8+) cgroup2 is used . This leads to missing "/sys/fs/cgroup/memory/memory.usage_in_bytes" on the pods and the metric 'kadalu_memory_usage_in_bytes' is presented as "-1".

**To Reproduce**
1. Enable user workload metrics 
2. Apply the following manifest:
```
---
kind: Namespace
apiVersion: v1
metadata:
  name: kadalu
  labels:
    openshift.io/cluster-monitoring: "true"
---
kind: Service
apiVersion: v1
metadata:
  name: operator
  namespace: kadalu
  labels:
    name: kadalu
spec:
  selector:
    name: kadalu
  ports:
    - name: kadalu-metrics
      protocol: TCP
      port: 8050
      targetPort: 8050
---
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  name: kadalu-metrics
  namespace: kadalu
  labels:
    name: kadalu
spec:
  endpoints:
    - path: /metrics
      port: kadalu-metrics
      scheme: http
      bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
      tlsConfig:
        insecureSkipVerify: true
  selector:
    matchLabels:
      name: kadalu
```
3. Set necessary permissions:
```
oc policy add-role-to-user kadalu-operator system:serviceaccount:openshift-monitoring:prometheus-k8s -n kadalu
```

**Expected behavior**
kadalu_memory_usage_in_bytes should present actual data or be removed.

**Actual behavior**
The metrics is logging in Openshift/OKD with value of "-1".

**Environment:**
1. 
```
$ kubectl-kadalu version
kubectl-kadalu plugin: 0.8.14
kadalu pod(s) versions
pod/kadalu-csi-nodeplugin-j9twl: 0.8.14
pod/kadalu-csi-nodeplugin-nrhm9: 0.8.14
pod/kadalu-csi-provisioner-0: 0.8.14
pod/operator-54696d985f-5wgdk: 0.8.14
pod/server-gluster-0-0: 0.8.14
pod/server-gluster-1-0: 0.8.14
pod/server-gluster-2-0: 0.8.14
```
2. 
```
$ oc adm upgrade
Cluster version is 4.10.0-0.okd-2022-04-23-131357
```

### PR DESCRIPTION
**Pull request by hunter86bg**
Signed-off-by: hunter86bg <hunter86_bg@yahoo.com>

**What this PR does / why we need it**:
Fixes cpu/memory metrics on cgroup2 pods

**Which issue(s) this PR fixes**:
Fixes #810

### CODE REVIEW

**Comment by hunter86bg** [comment]
@leelavg ,

any idea how  to fix the pylint ?

**Comment by amarts** [comment]
One way was to make multiple methods (functions) for such decisions, so number of branches in a single method would have reduced.  Considering this would fix an issue in monitoring, I prefer to get this in now, and optimize later.

@vatsa287 please keep this in backlog.

**Review by amarts — COMMENTED** [review]
LGTM

File: csi/exporter.py [hunk 1]
```diff
@@ -25,15 +26,28 @@ def metrics():
...
-    memory_usage_file_path = '/sys/fs/cgroup/memory/memory.usage_in_bytes'
-    if os.path.exists(memory_usage_file_path):
-        with open(memory_usage_file_path, 'r') as memory_fd:
+    # CPU (cgroup2)
+    if os.path.exists('/sys/fs/cgroup/cpu.stat'):
+        with open('/sys/fs/cgroup/cpu.stat', 'r') as cpu_fd:
+            for line in cpu_fd: # loop over all lines till we find usage_usec
+                if re.search('usage_usec', line):
+                    # Convert from usec (microseconds) to nanoseconds
+                    cpu_usage_in_nanoseconds = int(line.split(' ')[1]) * 1000
```
Comment (leelavg) [inline csi/exporter.py:35]: - can be taken in backlog
``` python
if line.startswith('usage_usec'):
  cpu_usage_in_nanoseconds = int(line.split()[-1]) * 1000
  break
```

**Review by leelavg — APPROVED** [review]

**Pull request merged**